### ISSUE TITLE
Could you explain what are vertex matrix and context matrix?

### ISSUE THREAD

**Issue by HenryYihengXu** [DeepGraphLearning/graphvite#41]
I'm reading your paper and have a question about section 3.2. Could you explain what are the vertex matrix and context matrix? Are they simply the source and the destination of edges?

**Comment by KiddoZhu** [comment]
Yes. Every node has a vertex and a context embedding. Vertex embedding is used when it acts as a source node, and context embedding is used for destination node.

**Comment by HenryYihengXu** [comment]
Are you saying each node will have two embedding vectors? So the number of embeddings will be twice as the number of nodes?

I don't quite understand the figure 2 in your Graphvite paper. In the figure, you have 4 GPUs and you divide the sample pool into 4x4 grids. Grid (i,j) contains edges with source in vertex i and dest in context j right? So what is the episode size here? Does it indicate that it takes many episodes to compute a single grid?

**Comment by HenryYihengXu** [comment]
And by your sampling method, each grid could contain different numbers of edges right?

**Comment by KiddoZhu** [comment]
For the embedding, yes.

The episode size is the **number of batch** in a grid, e.g. 500 for Youtube dataset. Each grid contains **episode size * batch size** samples. The is mainly because a sufficiently large grid is needed to reduce swap of embedding parameters across GPUs.

**Comment by razrLeLe** [comment]
@KiddoZhu  What's the purpose of setting two embedding vectors for each node?

**Comment by KiddoZhu** [comment]
One vector is called vector representation, the other is called context representation. This is mainly borrowed from the [distributional hypothesis](https://en.wikipedia.org/wiki/Distributional_semantics#Distributional_hypothesis) in NLP.

Empirically it works better than tied vertex and context parameters.

**Comment by HJW3536** [comment]
> One vector is called vertex representation, the other is called context representation. This is mainly borrowed from the [distributional hypothesis](https://en.wikipedia.org/wiki/Distributional_semantics#Distributional_hypothesis) in NLP.
> 
> Empirically it works better than tied vertex and context parameters.

hi,when I use them, how do I choose vertex embeddings or context embeddings?

**Comment by chi2liu** [comment]
> > One vector is called vertex representation, the other is called context representation. This is mainly borrowed from the [distributional hypothesis](https://en.wikipedia.org/wiki/Distributional_semantics#Distributional_hypothesis) in NLP.
> > Empirically it works better than tied vertex and context parameters.
> 
> hi,when I use them, how do I choose vertex embeddings or context embeddings?

vertex embeddings is better
